### PR TITLE
Pass sibling to `Renderer.mount`, fix update order

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Resolves, but adds no tests cases to the test suite for #294. See the issue for the detailed description of the problem.

I will add end-to-end tests for this in future PRs.

I've tested these cases manually so far:

```swift
struct Choice: View {
  @State private var choice = false

  var body: some View {
    HStack {
      Button("Trigger") {
        choice.toggle()
      }
      if choice {
        Group {
          Text("true")
          Text("true")
        }
      } else {
        VStack {
          Text("false")
        }
      }
      Text("end")
    }
  }
}
```

Note the `Group` view with multiple children in this one, it uncovered required checks for `GroupView` conformance.

Also tested these more simple cases:

```swift
struct Choice: View {
  @State private var choice = false

  var body: some View {
    HStack {
      Button("Trigger") {
        choice.toggle()
      }
      if choice {
        Group {
          // single child
          Text("true")
        }
      } else {
        VStack {
          Text("false")
        }
      }
      Text("end")
    }
  }
}
```

and

```swift
struct Choice: View {
  @State private var choice = false

  var body: some View {
    HStack {
      Button("Trigger") {
        choice.toggle()
      }
      if choice {
        // single child, no nesting
        Text("true")
      } else {
        VStack {
          Text("false")
        }
      }
      Text("end")
    }
  }
}
```

### CODE REVIEW

File: Sources/TokamakCore/MountedViews/MountedApp.swift [hunk 1]
```diff
@@ -22,12 +22,12 @@ import Runtime
 // is the computed content of the specified `Scene`, instead of having child
 // `View`s
 final class MountedApp<R: Renderer>: MountedCompositeElement<R> {
-  override func mount(with reconciler: StackReconciler<R>) {
+  override func mount(before _: R.TargetType? = nil, with reconciler: StackReconciler<R>) {
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/MountedViews/MountedApp.swift:25]: `App` elements have no siblings

File: Sources/TokamakCore/Renderer.swift [hunk 2]
```diff
@@ -31,9 +31,6 @@ public protocol Renderer: AnyObject {
    */
   associatedtype TargetType: Target
 
-  /// Reconciler instance used by this renderer.
-  var reconciler: StackReconciler<Self>? { get }
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/Renderer.swift:35]: It wasn't referenced anywhere, so no need to have it as a requirement. It also allowed making this property completely `private` in `Renderer` implementations.

File: Sources/TokamakDOM/DOMRenderer.swift [hunk 3]
```diff
@@ -71,7 +71,7 @@ func appendRootStyle(_ rootNode: JSObject) {
 }
 
 final class DOMRenderer: Renderer {
-  private(set) var reconciler: StackReconciler<DOMRenderer>?
+  private var reconciler: StackReconciler<DOMRenderer>?
```
Comment (MaxDesiatov) [inline Sources/TokamakDOM/DOMRenderer.swift:74]: Still keeping the property implementation itself around, as we need a non-zero retain count on it. The reconciler is really what drives the rendering process.

**Review by carson-katri — APPROVED** [review]
Looks good! The original issue was a great catch 😅

**Pull request merged**